pkg/controller/infrastructure/actuator_reconcile.go

```diff
@@ -123,14 +123,18 @@ func generateTerraformInfraConfig(ctx context.Context, infrastructure *extension
 			return nil, err
 		}
 		existingVpcID := *infrastructureConfig.Networks.VPC.ID
+		if err := awsClient.VerifyVPCAttributes(ctx, existingVpcID); err != nil {
```

Comment:
BTW, what happens when a user specifies a non-existing vpc-id? I guess, this will fail here, right?
Would it make sense to check if the vpc exists explicitly so we can return a meaningful error message?

Reply:
The error message of the `Infrastructure` controller will be

```
failed to generate Terraform config: InvalidVpcID.NotFound: The vpc ID 'vpc-0897467802289706f' does not exist\n\tstatus code: 400, request id: 443e0c6e-b212-4a49-b3a7-35198f322ba9"
```

Reply:
Well, that's meaningful enough :)
Thanks for checking!